java-frontend/src/main/java/org/sonar/java/se/checks/FlowComputation.java

```diff
@@ -117,14 +122,51 @@ private void run(@Nullable final ExplodedGraph.Node currentNode, @Nullable final
     final ExplodedGraph.Node parent = currentNode.parent();
     learnedConstraints.stream()
       .filter(addToFlow.and(Objects::nonNull))
-      .forEach(lc -> flow.add(location(parent, learnedConstraintMessage(lc, parent))));
+      .forEach(lc -> {
+        flow.add(location(parent, learnedConstraintMessage(lc, currentNode, parent)));
+        if (parent.programPoint.syntaxTree().is(Tree.Kind.METHOD_INVOCATION)) {
+          List<MethodYield> usedMethodYields = currentNode.usedMethodYields(parent);
+          if (!usedMethodYields.isEmpty()) {
+            flow.addAll(flowFromMethodInvocation(usedMethodYields, parent));
+          }
+        }
+      });
+
     return learnedConstraints;
   }
 
-  private static String learnedConstraintMessage(Constraint lc, ExplodedGraph.Node currentNode) {
-    Tree nodeTree = currentNode.programPoint.syntaxTree();
+  private List<JavaFileScannerContext.Location> flowFromMethodInvocation(List<MethodYield> usedMethodYields, ExplodedGraph.Node parentNode) {
+    // TODO use other yields as alternative flow?
+    MethodYield firstMethodYield = usedMethodYields.get(0);
+    int argumentIndex = correspondingArgumentIndex(symbolicValue, parentNode);
+    return firstMethodYield.flow(argumentIndex);
+  }
+
+  private static int correspondingArgumentIndex(SymbolicValue candidate, ExplodedGraph.Node invocationNode) {
+    MethodInvocationTree mit = (MethodInvocationTree) invocationNode.programPoint.syntaxTree();
+    List<SymbolicValue> arguments = argumentsUsedForMethodInvocation(invocationNode, mit);
+    return arguments.indexOf(candidate);
+  }
+
+  private static List<SymbolicValue> argumentsUsedForMethodInvocation(ExplodedGraph.Node invocationNode, MethodInvocationTree mit) {
+    List<SymbolicValue> values = new ArrayList<>(invocationNode.programState.peekValues(mit.arguments().size()));
+    Collections.reverse(values);
+    return values;
+  }
+
+  private static String learnedConstraintMessage(Constraint lc, ExplodedGraph.Node currentNode, ExplodedGraph.Node parent) {
+    Tree nodeTree = parent.programPoint.syntaxTree();
     String name = SyntaxTreeNameFinder.getName(nodeTree);
     if (nodeTree.is(Tree.Kind.METHOD_INVOCATION)) {
+      List<MethodYield> usedMethodYields = currentNode.usedMethodYields(parent);
+      if (!usedMethodYields.isEmpty()) {
+        // TODO use other yields as alternative flows ?
+        Type exceptionType = usedMethodYields.get(0).exceptionType();
+        // Only considering exceptional flows
+        if (exceptionType != null) {
```

Comment:
Wondering about this limitation in fact. Maybe to be addressed in another ticket.